### PR TITLE
[FLINK-4422] Convert all time interval measurements to System.nanoTime() - flink-clients

### PR DESCRIPTION
**Pull request by jinmingjian**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: flink-clients/src/test/java/org/apache/flink/client/program/ClientConnectionTest.java [hunk 1]
```diff
@@ -115,13 +115,13 @@ public void run() {
 
 			try {
 				// wait until the caller is successful, for at most the given time
-				long now = System.currentTimeMillis();
-				long deadline = now + MAX_DELAY;
+				long now = System.nanoTime();
+				long deadline = now + MAX_DELAY*1000000;
```
Comment (zentol) [inline flink-clients/src/test/java/org/apache/flink/client/program/ClientConnectionTest.java]: missing spaces around ```*```

Comment (StephanEwen) [inline flink-clients/src/test/java/org/apache/flink/client/program/ClientConnectionTest.java]: I would suggest to express the million literal as `1_000_000` - that is easier to parse for the reader, making it easier to spot errors.

Comment (jinmingjian) [inline flink-clients/src/test/java/org/apache/flink/client/program/ClientConnectionTest.java]: agree, although I commonly use 1000_000 in fact:) I planed to use [this detailed idea style](https://raw.githubusercontent.com/dawidwys/flink/bee981be8aa83225a8e3a736700eae445d79cece/tools/FlinkCodeStyle.xml) after digging today's discussions in mail list. Is this OK?:)

**Comment by StephanEwen** [comment]
Looks good, +1 to merge

Optional comment: I have seen that developers get confused when working with the code whether a `long` refers to a "millisecond" timestamp or to a "nanosecond" timestamp.

One way to solve that is to put that into the variable names (like `MAX_DELAY_MILLIS` or `deadlineNanos`). Another way is to use only milliseconds everywhere. Something like https://github.com/apache/flink/blob/master/flink-runtime/src/main/java/org/apache/flink/runtime/util/clock/Clock.java can help to get the milliseconds in a more readable way than using nanoTime and multiplying.

**Pull request closed without merging**